control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -89,6 +89,26 @@ type NamespacedReconciler struct {
 }
 
 func (r *NamespacedReconciler) ReconcileKind(ctx context.Context, broker *eventing.Broker) reconciler.Event {
+	if broker.Spec.Config != nil && broker.Spec.Config.Namespace != "" && broker.Spec.Config.Namespace != broker.Namespace {
+		return propagateErrorCondition(broker,
+			fmt.Errorf("broker with %q class need the config in the same namespace with the broker. broker.spec.config.namespace=%q, broker.namespace=%q",
+				kafka.NamespacedBrokerClass, broker.Spec.Config.Namespace, broker.Namespace))
+	}
+
+	br := r.createReconcilerForBrokerInstance(broker)
+
+	event := r.reconcileDataPlane(ctx, br, broker)
+	if event != nil {
+		return event
+	}
+
+	return br.ReconcileKind(ctx, broker)
+}
+
+// reconcileDataPlane reconciles the data plane in the namespace of the given broker.
+// A lock is acquired for the namespace of the broker to ensure that the creation of the data plane is not done
+// at the same time with the deletion of the data plane.
+func (r *NamespacedReconciler) reconcileDataPlane(ctx context.Context, br *Reconciler, broker *eventing.Broker) reconciler.Event {
```

Comment:
is `br` used somewhere?

Reply:
Removed it already. You were checking old commit I guess